### PR TITLE
Add Support for XDG Base Directories

### PR DESCRIPTION
**Pull request by olekw**
Implements the following blueprint: "Conform to the XDG base directory structure" at https://blueprints.launchpad.net/ember/+spec/ember-xdg-basedirectories.

### CODE REVIEW

File: ember.in [hunk 1]
```diff
@@ -6,27 +6,78 @@ orig_args=$@
...
+  --legacyhome)
+    legacyhome=1
+    ;;
   esac
   shift
 done
 
+# Migrate from legacy home directories to those specified by the XDG Base Directory Specification
+# Only migrate if legacy directory exists and user does not request it kept
+if [ $legacyhome = 0 ] && [ -d "$HOME/.ember" ] ; then
```
Comment (erikogenvik) [inline ember.in]: Good work here!

File: src/services/server/LoggedInState.cpp [hunk 2]
```diff
@@ -67,7 +67,7 @@ LoggedInState::~LoggedInState()
 void LoggedInState::checkTransfer()
 {
 	TransferInfoStringSerializer serializer;
-	std::string teleportFilePath(EmberServices::getSingleton().getConfigService().getHomeDirectory() + "/teleports");
+	std::string teleportFilePath(EmberServices::getSingleton().getConfigService().getHomeDirectory(BaseDirType_CACHE) + "teleports");
```
Comment (erikogenvik) [inline src/services/server/LoggedInState.cpp]: This shouldn't be in cache.
The transfer information is crucial for letting a player follow their avatar as it moves between servers. Think of it like a password, or a key. If it's lost the avatar is lost also.

File: ember.in (hunk 1 above)
Comment (olekw) [inline ember.in]: Thanks. :)

File: src/services/server/LoggedInState.cpp (hunk 2 above)
Comment (olekw) [inline src/services/server/LoggedInState.cpp]: Ah, ok. Good to know!

**Pull request merged**